File: Arrays: Subarrays/printAll.py
def printSubarray(A, start, end):
    for i in range(start, end + 1):
        print(A[i], end=" ")
    print()


def printAllSubarrays(A):
    n = len(A)

    for i in range(n):
        for j in range(i, n):
            printSubarray(A, i, j)


printAllSubarrays([2, 7, 3, 4])

File: Arrays: Subarrays/PrintAll.java
public class PrintAll {
    void printSubarray(int []A, int start, int end) {
        for (int i = start; i<=end;i++) {
            System.out.print(A[i] + " ");
        }
        System.out.println();
    }

    void printAllSubarrays(int []A) {
        int n = A.length;
        for (int i = 0; i<n;i++) {
            for (int j = i; j<n;j++) {
                printSubarray(A, i, j);
            }
        }
    }

    public static void main(String[] args) {
        PrintAll p = new PrintAll();
        int []A = {1, 2, 3, 4, 5};
        p.printAllSubarrays(A);
    }
}
